spring-session/src/main/java/org/springframework/session/data/hazelcast/config/annotation/web/http/HazelcastHttpSessionConfiguration.java

```diff
@@ -0,0 +1,124 @@
+ * "springSessionRepositoryFilter". In order to use this a single
+ * {@link HazelcastInstance} must be exposed as a Bean.
+ * 
+ * @author Tommy Ludwig
+ * @since 1.1
+ * @see EnableHazelcastHttpSession
+ */
+@Configuration
+public class HazelcastHttpSessionConfiguration extends SpringHttpSessionConfiguration implements ImportAware, BeanClassLoaderAware {
+	
+	/** This is the magic value to use if you do not want this configuration 
+	 * overriding the maxIdleSeconds value for the Map backing the session data. */
+	public static final String DO_NOT_CONFIGURE_INACTIVE_INTERVAL_STRING = "";
+	
+	private ClassLoader beanClassLoader;
+	
+	private Integer maxInactiveIntervalInSeconds = 1800;
+	
+	private String sessionMapName = "spring:session:sessions";
+
+	@Bean
+	public SessionRepository<ExpiringSession> sessionRepository(HazelcastInstance hazelcastInstance) {
+		configureSessionMap(hazelcastInstance);
+		Map<String,ExpiringSession> sessions = hazelcastInstance.getMap(sessionMapName);
+
+		SessionRepository<ExpiringSession> sessionRepository = new MapSessionRepository(sessions);
+		
+		return sessionRepository;
+	}
+	
+	/**
+	 * Make a {@link MapConfig} for the given sessionMapName if one does not exist.
+	 * Ensure that maxIdleSeconds is set to maxInactiveIntervalInSeconds for proper session expiration.
+	 * 
+	 * @param hazelcastInstance the {@link HazelcastInstance} to configure
+	 */
+	private void configureSessionMap(HazelcastInstance hazelcastInstance) {
+		MapConfig sessionMapConfig = hazelcastInstance.getConfig().getMapConfig(sessionMapName);
+		if (this.maxInactiveIntervalInSeconds != null) {
+			sessionMapConfig.setMaxIdleSeconds(this.maxInactiveIntervalInSeconds);
+		}
+	}
+
+	public void setImportMetadata(AnnotationMetadata importMetadata) {
+		Map<String, Object> enableAttrMap = importMetadata.getAnnotationAttributes(EnableHazelcastHttpSession.class.getName());
+		AnnotationAttributes enableAttrs = AnnotationAttributes.fromMap(enableAttrMap);
+		if (enableAttrs == null) {
+			// search parent classes
+			Class<?> currentClass = ClassUtils.resolveClassName(importMetadata.getClassName(), beanClassLoader);
+			for (Class<?> classToInspect = currentClass; classToInspect != null; classToInspect = classToInspect.getSuperclass()) {
+				EnableHazelcastHttpSession enableHazelcastHttpSessionAnnotation = AnnotationUtils.findAnnotation(classToInspect, EnableHazelcastHttpSession.class);
+				if (enableHazelcastHttpSessionAnnotation == null) {
+					continue;
+				}
+				enableAttrMap = AnnotationUtils
+						.getAnnotationAttributes(enableHazelcastHttpSessionAnnotation);
+				enableAttrs = AnnotationAttributes.fromMap(enableAttrMap);
+			}
+		}
+		
+		if (DO_NOT_CONFIGURE_INACTIVE_INTERVAL_STRING.equals(enableAttrs.getString("maxInactiveIntervalInSeconds"))) {
+			this.maxInactiveIntervalInSeconds = null;
+		} else {
+			this.maxInactiveIntervalInSeconds = enableAttrs.getNumber("maxInactiveIntervalInSeconds");
```

Comment:
If something other than a `Number` is given, the application will fail on startup with a straightforward message, which I figured was sufficient exception handling. Let me know if you want anything else.